WireDomain/Sources/WireDomain/Repositories/Conversations/Models/SystemMessage.swift

```diff
@@ -0,0 +1,59 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import WireDataModel
+
+public struct SystemMessage {
+    let type: ZMSystemMessageType
+    let sender: ZMUser
+    let users: Set<ZMUser>?
+    let addedUsers: Set<ZMUser>
+    let clients: Set<WireDataModel.UserClient>?
+    let timestamp: Date
+    let duration: TimeInterval?
+    let messageTimer: Double?
+    let relevantForStatus: Bool
+    let removedReason: ZMParticipantsRemovedReason
+    let domains: [String]?
```

Comment:
Maybe it's not a good idea to have references managed objects here since it's not clear to the outside that `SystemMessage` is now bound to a certain queue.

Reply:
Agreed, this model was removed in the following ongoing [PR](https://github.com/wireapp/wire-ios/pull/2096).